banners/english/banner_ctrl.ts

```diff
@@ -32,6 +32,7 @@ const mediaWiki = new WindowMediaWiki();
 const page = new PageWPORG( mediaWiki, ( new SkinFactory( mediaWiki ) ).getSkin(), new WindowSizeIssueChecker( 800 ) );
 const impressionCount = new LocalImpressionCount( page.getTracking().keyword );
 const tracker = new LegacyTrackerWPORG( mediaWiki, page.getTracking().keyword, eventMappings );
+const remainingImpressions = Math.max( page.getMaxBannerImpressions( 'english' ) - impressionCount.overallCountIncremented, 0 );
```

Comment:
You could make the entry points a bit nicer and readable by adding `page.getRemainingImpressions(impressionCount.overallCountIncremented)` and moving the `Math.max` and `getMaxBannerImpressions` to `page`.

Reply:
I'm reluctant to do that because it would create a soft 2 way dependency between `page` and `impressionCount`